overlord/snapstate/snapmgr.go

```diff
@@ -381,10 +393,75 @@ func (m *SnapManager) blockedTask(cand *state.Task, running []*state.Task) bool
 	return false
 }
 
+// ensureRefreshes ensures that we refresh all installed snaps periodically
+func (m *SnapManager) ensureRefreshes() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	var nextRefresh time.Time
+	err := m.state.Get("next-auto-refresh-time", &nextRefresh)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+
+	// time to refresh?
+	if !nextRefresh.IsZero() && time.Now().Before(nextRefresh) {
```

Comment:
I suppose we shouldn't trigger a refresh before we are seeded?

it seems that if the refresh fails we will try immediately again instead of waiting a bit? because schedule-next-refresh is invoked only if everything is ok?